src/js/view/week/weekdayInWeek.js

```diff
@@ -35,21 +41,55 @@ util.inherit(WeekdayInWeek, Weekday);
 WeekdayInWeek.prototype.render = function(viewModel) {
     var opt = this.options,
         container = this.container,
-        baseViewModel = this.getBaseViewModel(viewModel),
-        maxScheduleInDay = 0;
-
-    baseViewModel.matrices = opt.getViewModelFunc(viewModel);
+        matrices = opt.getViewModelFunc(viewModel),
+        renderLimitIdx = this._getRenderLimitIndex();
+    var maxScheduleInDay, baseViewModel, panelHeight, visibleScheduleCount;
 
     maxScheduleInDay = mmax.apply(
         null,
-        util.map(baseViewModel.matrices, function(matrix) {
+        util.map(matrices, function(matrix) {
             return Math.max.apply(null, util.map(matrix, function(row) {
                 return row.length;
             }));
         })
     );
 
-    baseViewModel.minHeight = this._getMinHeight(maxScheduleInDay);
+    if (this.collapsed) {
+        panelHeight = this.aboutMe.forcedLayout ? this.getViewBound().height : opt.panelHeights[0];
+        visibleScheduleCount = Math.floor(panelHeight / (opt.scheduleHeight + opt.scheduleGutter));
+
+        if (maxScheduleInDay > visibleScheduleCount) {
+            viewModel.exceedDate = this.getExceedDate(
+                renderLimitIdx - 1, viewModel.schedulesInDateRange[this.aboutMe.panelName]
+            );
+
+            matrices = matrices.map(function(matrix) {
+                return matrix.map(function(row) {
+                    if (row.length >= visibleScheduleCount) {
+                        return row.filter(function(item) {
+                            return item.top < visibleScheduleCount - 1;
+                        });
+                    }
+
+                    return row;
+                });
+            });
+        } else {
+            viewModel.exceedDate = this.getExceedDate(
+                renderLimitIdx, viewModel.schedulesInDateRange[this.aboutMe.panelName]
+            );
+        }
+        baseViewModel = this.getBaseViewModel(viewModel);
+        baseViewModel.fixed = 'fixed';
+        this.fire('sendMaxScheduleInDay', maxScheduleInDay);
```

Comment:
To fire events in `render` function which is frequently called is not good for performance. Let's rethink this flow.